px4_ros2_cpp/include/px4_ros2/components/mode_executor.hpp

```diff
@@ -35,6 +35,7 @@ class ModeExecutorBase
   struct Settings
   {
     bool activate_immediately{false};             ///< If set activate the mode (and executor) immediately. Only use this for fully autonomous executors that also arm the vehicle
+    bool is_allowed_to_arm{false};                ///< If set, the executor is allowed to arm the vehicle
```

Comment:
I would combine these into an enum, since they are related:
```cpp
enum class Activation {
  ActivateOnlyWhenArmed, ///< Only activate the executor when armed (and selected)
  ActivateAlways,  ///< Allow the executor to always be activated (so it can arm the vehicle)
  ActivateImmediately, ///< Activate the mode and executor immediately after registration. Only use this for fully autonomous executors that also arm the vehicle
};
```